domain/application/state/charm_test.go

```diff
@@ -950,37 +958,43 @@ INSERT INTO charm_storage_property (
 	})
 	c.Assert(err, jc.ErrorIsNil)
 
+	expectedStorage := map[string]charm.Storage{
+		"foo": {
+			Name:        "bar",
+			Type:        charm.StorageFilesystem,
+			Description: "description 1",
+			Shared:      true,
+			ReadOnly:    true,
+			CountMin:    1,
+			CountMax:    2,
+			MinimumSize: 3,
+			Location:    "/tmp",
+			Properties:  []string{"alpha", "beta", "beta"},
+		},
+		"fred": {
+			Name:        "baz",
+			Type:        charm.StorageBlock,
+			Description: "description 2",
+			Shared:      false,
+			ReadOnly:    false,
+			CountMin:    4,
+			CountMax:    5,
+			MinimumSize: 6,
+			Location:    "/var/mount",
+		},
+	}
+
 	metadata, err := st.GetCharmMetadata(context.Background(), id)
 	c.Assert(err, jc.ErrorIsNil)
 
 	assertCharmMetadata(c, metadata, func() charm.Metadata {
-		expected.Storage = map[string]charm.Storage{
-			"foo": {
-				Name:        "bar",
-				Type:        charm.StorageFilesystem,
-				Description: "description 1",
-				Shared:      true,
-				ReadOnly:    true,
-				CountMin:    1,
-				CountMax:    2,
-				MinimumSize: 3,
-				Location:    "/tmp",
-				Properties:  []string{"alpha", "beta", "beta"},
-			},
-			"fred": {
-				Name:        "baz",
-				Type:        charm.StorageBlock,
-				Description: "description 2",
-				Shared:      false,
-				ReadOnly:    false,
-				CountMin:    4,
-				CountMax:    5,
-				MinimumSize: 6,
-				Location:    "/var/mount",
-			},
-		}
+		expected.Storage = expectedStorage
 		return expected
 	})
+
+	storage, err := st.GetCharmMetadataStorage(context.Background(), id)
```

Comment:
Can these be put into seperate functions for testing? The preamble for this could just be put into a function that all the individual tests call. That'd give a clearer signal if any test starts failing.

Reply:
Could do, but that would lead to a lot of code duplication in the tests. I'd rather keep them like this tbh